ros_bt_py/ros_bt_py/ros_nodes/service.py

```diff
@@ -79,11 +78,73 @@ class ServiceInput(Leaf):
     data.
     """
 
+    _request_type: type
+    _response_type: type
+    _service_client: Optional[Client] = None
+
+    def __init__(
+        self,
+        options: Optional[Dict] = None,
+        debug_manager: Optional[DebugManager] = None,
+        subtree_manager: Optional[SubtreeManager] = None,
+        name: Optional[str] = None,
+        ros_node: Optional[Node] = None,
+        succeed_always: bool = False,
+        simulate_tick: bool = False,
+    ) -> None:
+        super().__init__(
+            options=options,
+            debug_manager=debug_manager,
+            subtree_manager=subtree_manager,
+            name=name,
+            ros_node=ros_node,
+            succeed_always=succeed_always,
+            simulate_tick=simulate_tick,
+        )
+
+        node_inputs = {}
+        node_outputs = {}
+        try:
+            self._request_type = getattr(self.options["service_type"], "Request")
+            if inspect.isclass(self._request_type):
+                msg = self._request_type()
+                for field in msg._fields_and_field_types:
+                    node_inputs[field] = type(getattr(msg, field))
+                self.passthrough = False
+            else:
+                node_inputs["in"] = self.options["service_type"]
```

Comment:
Does this default make sense? Why would we not fail if the request_type is not a Message Type?